### PR TITLE
Add users to a project from prereg form

### PR DESCRIPTION
**Pull request by atelic**
[OSF-4975]

### CODE REVIEW

File: website/static/js/contribAdder.js [hunk 1]
```diff
@@ -306,8 +308,14 @@ var AddContributorViewModel = oop.extend(Paginator, {
                 }),
                 node_ids: self.nodesToChange()
             }
-        ).done(function() {
-            window.location.reload();
+        ).done(function(response) {
+            if (self.async) {
+                $('.modal').modal('hide');
+                $osf.unblock();
+                self.callback(response);
```
Comment (samchrisinger) [inline website/static/js/contribAdder.js]: This needs a `if(self.callback)`

File: website/static/js/contribAdder.js [hunk 2]
```diff
@@ -306,8 +308,14 @@ var AddContributorViewModel = oop.extend(Paginator, {
                 }),
                 node_ids: self.nodesToChange()
             }
-        ).done(function() {
-            window.location.reload();
+        ).done(function(response) {
+            if (self.async) {
+                $('.modal').modal('hide');
```
Comment (samchrisinger) [inline website/static/js/contribAdder.js:313]: Use `bootbox.hideAll`

Comment (atelic) [inline website/static/js/contribAdder.js:313]: The modal is not a bootbox dialog so this will not work.

File: website/static/js/contribAdder.js [hunk 3]
```diff
@@ -37,6 +37,8 @@ var AddContributorViewModel = oop.extend(Paginator, {
         self.nodeApiUrl = '/api/v1/project/' + self.nodeId + '/';
         self.parentId = parentId;
         self.parentTitle = parentTitle;
+        self.async = options.async;
```
Comment (samchrisinger) [inline website/static/js/contribAdder.js:40]: This needs `options = options || {};`

File: website/static/js/pages/register-page.js [hunk 4]
```diff
@@ -17,35 +18,32 @@ $(function() {
...
-        schemas: '/api/v1/project/schemas/',
-        create: node.urls.api + 'drafts/',
-        submit: node.urls.api + 'drafts/{draft_pk}/submit/',
-        update: node.urls.api + 'drafts/{draft_pk}/',
-        get: node.urls.api + 'drafts/{draft_pk}/',
-        draftRegistrations: node.urls.web + 'registrations/'
-    }, null, true);
-
-    if (ctx.draft) { // if registering draft
+    // if registering draft
```
Comment (samchrisinger) [inline website/static/js/pages/register-page.js]: I refactored this block some and it looks like it's been rolled back. Any reason for this?

File: website/static/js/pages/sharing-page.js [hunk 5]
```diff
@@ -19,7 +19,8 @@ $('body').on('nodeLoad', function(event, data) {
             data.node.title,
             data.node.id,
             data.parent_node.id,
-            data.parent_node.title
+            data.parent_node.title,
+            {}
```
Comment (samchrisinger) [inline website/static/js/pages/sharing-page.js]: If you add the suggested fallback for `options` you can remove this argument

File: website/static/js/registrationEditorExtensions.js [hunk 6]
```diff
@@ -180,59 +181,45 @@ var Uploader = function(data) {
...
+    self.makeContributorsRequest = function() {
+        var self = this;
+        var contributorsUrl = window.contextVars.node.urls.api + 'get_contributors/';
+        return $.getJSON(contributorsUrl);
+    };
+    self.getContributors = function() {
+        var self = this;
+        return self.makeContributorsRequest()
+            .then(function(data) {
+                return $.map(
```
Comment (samchrisinger) [inline website/static/js/registrationEditorExtensions.js]: Very minor, but can you move this mapping function (used below as well), to a shared function?

File: website/static/js/contribAdder.js (hunk 2 above)
Comment (samchrisinger) [inline website/static/js/contribAdder.js:313]: :(

File: website/static/js/pages/register-page.js (hunk 4 above)
Comment (atelic) [inline website/static/js/pages/register-page.js]: I don't remember making those changes so it's probably just a mistake. Will fix.

**Comment by atelic** [comment]
@samchrisinger changes made

File: website/static/js/contribAdder.js (hunk 3 above)
Comment (samchrisinger) [inline website/static/js/contribAdder.js:40]: Can you please address or respond to this?

Comment (samchrisinger) [inline website/static/js/contribAdder.js:40]: NVM, I see now

**Pull request merged**